gnosis/safe-react
# Offer the user to switch/add MetaMask chain

## What it solves
Resolves #2455 and #2360.

## How this PR fixes it
On wallet connect, it checks if the network is matching. If not, it will offer the user to switch the chain.
If that fails, it will try to add a new chain config.

## How to test it
Case 1 – existing chains:
* Go to the Rinkeby site with the MM connected to mainnet

Case 2 – new chains:
* Connect MM to Rinkeby
* Remove your xDai config from MM
* Go to the xDai PR site (I'll temporarily enable it for this PR)

Case 3 – chains already matching
* Everything should work like before

## Screenshots
<img width="492" alt="Screenshot 2021-06-24 at 10 16 07" src="https://user-images.githubusercontent.com/381895/123227784-352a0180-d4d5-11eb-9735-5e44f139f617.png">

<img width="551" alt="Screenshot 2021-06-24 at 10 03 15" src="https://user-images.githubusercontent.com/381895/123225845-686b9100-d4d3-11eb-8c0a-6080e5095bc2.png">
